packages/toolpad-utils/src/fs.ts

```diff
@@ -55,7 +69,7 @@ export async function updateYamlFile(filePath: string, content: object) {
   const newContent = oldContent ? yamlOverwrite(oldContent, content) : yaml.stringify(content);
 
   if (newContent !== oldContent) {
```

Comment:
We can assume that in most cases `oldContent` will be formatted. That means `newContent` needs to be formatted before comparing. Otherwise it would write more often than necessary.